### DISCUSSION TITLE
Predefined coupled motion - Sloshing tank case

### DISCUSSION THREAD

**Discussion by VVictorGit** [DualSPHysics/DualSPHysics#216] [DUALSPHYSICS_V5.2]
I have used the predefined motion to excite the sloshing tank, either for translational or rotational movement. However, although the movements work separately, I couldn't find an option to have a coupled motion, for example: vertical translation with rotation. 

When I define the motion there is <begin mov='1', start ='0'/> . That means that the next motion is mov=2. As far as I saw when I give the same input sequence (mov=1 to both motions) is not taking into account the two motions in parallel. Let me know if there is a work around this feature, I would be interested to use the couple motion for the sloshing tank movement.

Is it working with newer versions, or it is the same situation? The version I used for my discussion was DualSPHysics v5.0. 

Thank you!

**Comment by alexbexe** [comment]
Dear Victor
Please take a look to examples/motion02, where a combination of two uniform rectilinear motion is used

Let us know if that works for you

Regards
Alex

**Reply by VVictorGit** [reply]
Dear Alex, 

thank you for recommendation! I noticed the file and I tried to replicate, but apparently I cannot get a combined motion. I attach the file in the archive to better understand the case that I am talking about. 

The case is CaseSloshingMotion_Def with external files to be read for the motion. The motions are rotational, and vertical, for a two-dimensional sloshing problem. I placed in the lines after duration of the motion, next='10' (for 10 seconds), but this gives me an error when I try to run the file. Without that command (next='10'), it works, but only the first motion (angular rotation) is taken into consideration into the simulation. Let me know if there is another way to introduce the combine motion. Thank you!

[Sloshing2D_CoupledMotion.zip](https://github.com/user-attachments/files/16429740/Sloshing2D_CoupledMotion.zip)

**Reply by VVictorGit** [reply]
Problem solved.

**Reply by AkbarAsis** [reply]
> Problem solved.

Dear Victor,
I have been working on a similar case to yours. Would you kindly explain how to solve coupled rotational and translational motion?

**Reply by VVictorGit** [reply]
Hello, 
I think I was doing a mistake in the reference, having the two motions with distinct references, thus not combining the motions, but rather having them one after the other, in sequence. However, I then used the same reference and it works in parallel. Here is the example I have used for the combined motion using v5.0

       <motion>
           <objreal ref='0'>
               <begin mov='1' start='0'/> 
                <mvrotfile id='1' duration='600' anglesunits='radians'> 
                 <file name='ExternalExcitationRotation.txt' /> 
                 <axisp1 x='0' y='-1' z='0' /> 
                 <axisp2 x='0' y='1' z='0' /> 
               </mvrotfile> 
               <begin mov='2' start='0'/> 
                <mvfile id='2' duration='600'>
                   <file name='ExternalExcitationVertical.txt' fields='2' fieldtime='0' fieldz='1' />
               </mvfile>
           </objreal>
       </motion>



Hope it helps.

Best regards,
Victor

**Reply by AkbarAsis** [reply]
Thank you for your clarification. I understand that using distinct references caused the motions to occur sequentially rather than being combined. Using the same reference to make them work in parallel makes sense.

Best regards,
Akbar
